packages/electrode-archetype-react-app/config/webpack/partial/pwa.js

```diff
@@ -106,14 +129,66 @@ module.exports = function () {
     var output = config.output;
     if (cacheConfig.importScripts) {
       var importScripts = cacheConfig.importScripts;
-      cacheConfig.importScripts = importScripts.map(getHashedPath);
+      cacheConfig.importScripts = process.env.WEBPACK_DEV === "true"
+        ? importScripts.map(getDevelopmentPath)
+        : importScripts.map(getHashedPath);
       entry = createEntryConfigFromScripts(importScripts, entry);
       output = {
         filename: "[name].[hash].js"
       };
     }
 
+
+    var plugins = [
+      new FaviconsWebpackPlugin({
+        logo: manifestConfig.logo,
+        emitStats: true,
+        inject: false,
+        background: manifestConfig.background,
+        title: manifestConfig.title,
+        statsFilename: manifestConfig.statsFilename,
+        icons: {
+          android: true,
+          appleIcon: true,
+          appleStartup: true,
+          favicons: true
+        }
+      }),
+      new AddManifestFieldsPlugin({
+        gcm_sender_id: manifestConfig.gcm_sender_id,
+        short_name: manifestConfig.short_name,
+        theme_color: manifestConfig.theme_color
+      }),
+      new SWPrecacheWebpackPlugin(cacheConfig)
+    ];
+
+    /**
+     * In dev we need to write the stats file to disk
+     * so we can properly read which chunk(s) need to be
+     * served. We write the stats file to our build artifacts
+     * folder, which is .build by default.
+     */
+    if (process.env.WEBPACK_DEV === "true") {
+      plugins.push(
+        new DiskPlugin({
+          output: {
+            path: path.resolve(
+              process.cwd(),
+              severConfig.buildArtifactsPath || ".build"
+            )
+          },
+          files: [{
+            asset: /\/stats.json$/,
+            output: {
+              filename: 'stats.json'
+            }
+          }]
+        })
+      );
+    }
+
     return mergeWebpackConfig(config, {
+      __wmlMultiBundle: typeof entry === 'object',
```

Comment:
That's one potential solution, but I'm not totally sure yet if that would break anything in the `pwa` partial.  It's probably safe, but I do think that removing the distinction is probably best as far as complexity goes, if we can manage it.